pkg/plugin/twinmaker/handler.go

```diff
@@ -476,11 +476,15 @@ func (s *twinMakerHandler) GetAlarms(ctx context.Context, query models.TwinMaker
 		}
 		failures = append(failures, newFailures...)
 		pValues = append(pValues, propertyReferences...)
-		if isLimited && len(pValues) >= maxResults {
-			if len(pValues) > maxResults {
-				pValues = pValues[:maxResults]
-			}
-			break
+		if isLimited {
+		    // update the queries' maxResults so we ask for less on the next iteration
+		    query.MaxResults = maxResults - len(pValues)
```

Comment:
You change query.MaxResults in subsequent calls so be careful how you mutate that field. It is always set to 0 so this won't work.
https://github.com/grafana/grafana-iot-twinmaker-app/pull/23/files#diff-4ddb8b65344e2d6f64d6e4b14dbb6a5f69a4961d7dfb6586620a47a65cafdd93R178

Reply:
this meant it was broken before and fixed now.

as i 'reset' query.MaxResults in this parent call.